### PR TITLE
add tcp support

### PR DESCRIPTION
**Pull request by Luap99**
see commits

Listen and forward via tcp as well not just udp

### CODE REVIEW

File: src/server/serve.rs [hunk 1]
```diff
@@ -144,16 +150,42 @@ async fn stop_and_start_threads<'a, Ip>(
...
+        let tcp_sock = match TcpListener::bind(addr).await {
+            Ok(s) => s,
+            Err(err) => {
+                errors.push(AardvarkError::wrap(
+                    format!("failed to bind tcp listener on {addr}"),
+                    err.into(),
+                ));
+                continue;
+            }
+        };
```
Comment (Luap99) [inline src/server/serve.rs:181]: @mheon here I am not sure what is best here

If we fail to bind udp or tcp we will skip that network and bind neither. I am not sure if it is worth to ignore tcp failures and still listen on udp or the other way around. I feel like this is not worth the extra complexity
WDYT?

**Comment by Luap99** [comment]
looks like the bind with ncat test is flaky (I need to add some sleep there even if that sucks)

File: src/server/serve.rs (hunk 1 above)
Comment (mheon) [reply src/server/serve.rs:181]: I think skipping the network is reasonable. Honestly, I would kind of prefer to shut down Aardvark altogether if we can't take the appropriate ports?

Comment (Luap99) [reply src/server/serve.rs:181]: Well if it is the first launch then we will exit with the right errors but if we just got reloaded via SIGHUP then we just log the error, at this point it is possible that other networks were able to bind.
Exiting means we bring down everything.

**Comment by Luap99** [comment]
Ok pushed another commits to do tcp forwarding as well

**Comment by TomSweeneyRedHat** [comment]
Addresses: https://issues.redhat.com/browse/RHEL-17270

**Pull request merged**